client/src/components/Tabs/README.md

```diff
@@ -33,3 +33,86 @@ clickable tab.
  * `tabClassName` (string): Class to use for the tab.
 
  _NOTE:_ For other properties, please refer to the [reactstrap Tabs](https://reactstrap.github.io/components/tabs/) documentation.
+
+# Interacting with Tabs from a child component
+
+Components inside a tab can retrieve information about their parent using these method:
+- `useTabContext` which is a hook returning the tab context for functional components
+- `injectTabContext` which injects the tab context as props in your class component
+- `useTabFirstShow` which is a hook that allows you to execute a callback when a tab is first shown.
+
+The `tabContext` returned by `useTabContext` and `injectTabContext` contains the following properties
+- `activeTab` (string): name of the active tab in the nearest tab set
+- `currentTab` (string): name of the tab the current component resides in
+- `isOnActiveTab` (boolean): whether the current component is in a tab that is currently visible.
+
+The `tabContext` will be `false` for components outside of a tab.
+ 
+
+## Example of `useTabContext`
+
+```jsx
+import React from 'react';
+import useTabContext from 'hooks/useTabContext';
+
+export function MyComponent() {
+  // useTabContext only works inside functinal components
+  const tabContext = useTabContext();
+  if (!tabContext) {
+    return <p>I'm not in a tab.</p>;
+  }
+
+  const {activeTab, currentTab, isOnActiveTab} = tabContext;
+  
+  return 
+    <p>
+        The {activeTab} tab is currently active.
+        I'm on the {currentTab} tab which is {isOnActiveTab ? 'visible' : 'invisible'}.
+    </p>; 
+}
+```
+
+## Example of `injectTabContext`
+
+```jsx
+import React, {Component} from 'react';
+import {injectTabContext} from 'hooks/useTabContext';
+
+class MyComponent extends Component {
+  render({tabContext}) {  
+    if (!tabContext) {
+      return <p>I'm not in a tab.</p>;
+    }
+    
+    const {activeTab, currentTab, isOnActiveTab} = tabContext;
+      
+    return 
+      <p>
+        The {activeTab} tab is currently active.
+        I'm on the {currentTab} tab which is {isOnActiveTab ? 'visible' : 'invisible'}.
+      </p>;
+  }
+}
+
+// `injectTabContext` is a higher-order-component (HOC). 
+// It can be applied to class and functional components.
```

Comment:
Seems like we'd only want to use it on class components though.  Kind of confusing saying we could use it on functional components, meanwhile we have 2 other functional specifically for functional components

Reply:
Some people prefer to use component composition. To be honest, the only reason I created `injectTabContext` was because `provideUsedOnData` is a class component. I would have preferred to just use hooks.

Reply:
How about

`// It is used to add context to class components.  It can also be used on functional components.`